packages/light-client/src/index.ts

```diff
@@ -529,44 +531,69 @@ export class Lightclient {
    *                     - current_sync_committee: period 0
    *                     - known next_sync_committee, signed by current_sync_committee
    */
-  private processSyncCommitteeUpdate(update: altair.LightClientUpdate): void {
+  private processLightClientUpdate(update: altair.LightClientUpdate): void {
     // Prevent registering updates for slots too far in the future
     const updateSlot = update.attestedHeader.slot;
+    const signatureSlot = update.signatureSlot;
+
+    if (signatureSlot < updateSlot) {
+      throw Error(
+        `signatureSlot: ${signatureSlot} where update is signed should not be before updateSlot: ${updateSlot}`
+      );
+    }
+
     if (updateSlot > slotWithFutureTolerance(this.config, this.genesisTime, MAX_CLOCK_DISPARITY_SEC)) {
       throw Error(`updateSlot ${updateSlot} is too far in the future, currentSlot ${this.currentSlot}`);
     }
 
     // Must not rollback periods, since the cache is bounded an older committee could evict the current committee
-    const updatePeriod = computeSyncPeriodAtSlot(updateSlot);
+    const attestedPeriod = computeSyncPeriodAtSlot(updateSlot);
+    const signaturePeriod = computeSyncPeriodAtSlot(signatureSlot);
     const minPeriod = Math.min(-Infinity, ...this.syncCommitteeByPeriod.keys());
-    if (updatePeriod < minPeriod) {
+    if (attestedPeriod < minPeriod) {
       throw Error(`update must not rollback existing committee at period ${minPeriod}`);
     }
 
-    const syncCommittee = this.syncCommitteeByPeriod.get(updatePeriod);
-    if (!syncCommittee) {
-      throw Error(`No SyncCommittee for period ${updatePeriod}`);
+    const newNextSyncCommitteeStats: LightclientUpdateStats = {
+      isFinalized: !isEmptyHeader(update.finalizedHeader),
+      participation: sumBits(update.syncAggregate.syncCommitteeBits),
+      slot: updateSlot,
+    };
+
+    let nextSyncCommittee: SyncCommitteeFast | undefined;
+    let signingSyncCommittee;
+
+    signingSyncCommittee = this.syncCommitteeByPeriod.get(signaturePeriod);
+
+    // in the scenario where only update for period is requested at the boundary,
+    // there won't have been an update to process that would have enabled next sync committee to be
+    // cached in this.syncCommitteeByPeriod, hence directly use one in update
+    if (isLastSlotInPeriod(updateSlot) && signingSyncCommittee === undefined) {
+      nextSyncCommittee = deserializeSyncCommittee(update.nextSyncCommittee);
+      signingSyncCommittee = {
+        ...newNextSyncCommitteeStats,
+        ...nextSyncCommittee,
+      };
+    }
```

Comment:
this is incorrect please remove this.
we cannot use data from the update to verify the authenticity of the update.

Reply:
The CI now fails a test with the following log:

```
Error sync {} Error: No syncCommittee for attested period 1 and signaturePeriod 2
```

This is because in the test case, the update for period 1 is set to happen at the boundary. This means that update would have been signed by sync committee in the next period (period 2) but that does not exist in the cache yet.